# Preliminary analysis of the Dataset

I just downloaded the dataset and want to determine the structure and the nature of variables in it

In [1]:
import pandas as pd

data = pd.read_json('dataset_airbnb-scraper_2023-06-18_04-11-45-096.json')
data.head(2)

url  \
0  https://www.airbnb.com/rooms/40751388   
1  https://www.airbnb.com/rooms/48645841   

                                                name  stars  numberOfGuests  \
0  Zen 2 bedroom apartment in the heart of El Pob...   4.96               4   
1              Private Jacuzzi Apartment in Provenza   4.64               6   

                         address                  roomType  \
0  Medellín, Antioquia, Colombia  Alojamiento entero: piso   
1  Medellín, Antioquia, Colombia  Alojamiento entero: piso   

                             location  \
0  {'lat': 6.20859, 'lng': -75.56774}   
1  {'lat': 6.20749, 'lng': -75.56672}   

                                             reviews  \
0  [{'author': {'firstName': 'Freddy', 'hasProfil...   
1  [{'author': {'firstName': 'Carlos', 'hasProfil...   

                                             pricing  \
0  {'rate': {'amount': 203, 'amountFormatted': '$...   
1  {'rate': {'amount': 179, 'amountFormatted': '$...   

                                              photos  ...  \
0  [{'caption': '', 'pictureUrl': 'https://a0.mus...  ...   
1  [{'caption': '', 'pictureUrl': 'https://a0.mus...  ...   

                                additionalHouseRules sortedReviews  \
0  - Check in 16:00 PM - Check out 11:00 AM.\n- A...            []   
1  •\tWe are Guest friendly (only one visitor per...            []   

   documentDisplayPictures  sections metadata  \
0                       []        []       {}   
1                       []        []       {}   

                         p3ImpressionId     idStr  \
0  ebcd798b-56b9-245b-9663-269436fe77ef  40751388   
1  fafbee62-1136-4ef7-55cb-49f3b0c97cf6  48645841   

  paidGrowthRemarketingListingIdsStr  \
0                                 []   
1                                 []   

                                       hostGuidebook accessibilityModule  
0                                                NaN                 NaN  
1  {'guidebookUrl': '/s/guidebooks?refinement_pat...                 NaN  

[2 rows x 85 columns]

In [2]:
data.columns

Index(['url', 'name', 'stars', 'numberOfGuests', 'address', 'roomType',
       'location', 'reviews', 'pricing', 'photos', 'primaryHost',
       'additionalHosts', 'isHostedBySuperhost', 'isAvailable', 'calendar',
       'occupancyPercentage', 'listingAmenities', 'rootAmenitySections',
       'seeAllAmenitySections', 'bathroomLabel', 'bedLabel', 'bedroomLabel',
       'hasHouseRules', 'id', 'isNewListing', 'license', 'listingRooms',
       'p3SummaryAddress', 'roomTypeCategory', 'sectionedDescription',
       'tierId', 'guestControls', 'hasSpecialOffer', 'maxNights', 'minNights',
       'priceDetails', 'hasWeWorkLocation', 'isBusinessTravelReady',
       'hometourRooms', 'hometourSections', 'descriptionLocale',
       'initialDescriptionAuthorType', 'localizedCheckInTimeWindow',
       'localizedCheckOutTime', 'city', 'country', 'countryCode',
       'hasHostGuidebook', 'hasLocalAttractions', 'neighborhoodCommunityTags',
       'state', 'paidGrowthRemarketingListingIds', 'hasCommercial

In [3]:
import numpy as np
for c in data.columns:
    if type(data[c][0]) == str:
        print('|'+c+'|str|')
    elif isinstance(data[c][0], np.floating):
        print('|'+c+'|float|')
    elif isinstance(data[c][0], np.int64):
        print('|'+c+'|int|')
    elif type(data[c][0]) == list:
        print('|'+c+'|list|')
    elif type(data[c][0]) == dict:
        print('|'+c+'|dict|')
    elif isinstance(data[c][0], np.bool_):
        print('|'+c+'|bool|')

|url|str|
|name|str|
|stars|float|
|numberOfGuests|int|
|address|str|
|roomType|str|
|location|dict|
|reviews|list|
|pricing|dict|
|photos|list|
|primaryHost|dict|
|additionalHosts|list|
|isHostedBySuperhost|bool|
|isAvailable|float|
|calendar|list|
|occupancyPercentage|float|
|listingAmenities|list|
|rootAmenitySections|list|
|seeAllAmenitySections|list|
|bathroomLabel|str|
|bedLabel|str|
|bedroomLabel|str|
|hasHouseRules|bool|
|id|int|
|isNewListing|bool|
|license|float|
|listingRooms|list|
|p3SummaryAddress|str|
|roomTypeCategory|str|
|sectionedDescription|dict|
|tierId|int|
|guestControls|dict|
|hasSpecialOffer|bool|
|maxNights|int|
|minNights|int|
|priceDetails|list|
|hasWeWorkLocation|bool|
|isBusinessTravelReady|bool|
|hometourRooms|list|
|hometourSections|list|
|descriptionLocale|str|
|initialDescriptionAuthorType|str|
|localizedCheckInTimeWindow|str|
|localizedCheckOutTime|str|
|city|str|
|country|str|
|countryCode|str|
|hasHostGuidebook|bool|
|hasLocalAttractions|bool|
|neigh

In [15]:
occup = data.occupancyPercentage[0]
print('Occupancy as calculated by Apify:', occup)

Occupancy as calculated by Apify: 25.63


In [22]:
## Making sure the occupancyPercentage variable is what I think it is

df = pd.DataFrame.from_dict(data.calendar[0])
#print( df[df.available == True].count() )
oc = df[df.available == False].count() * 100 / df.count()
assert round(occup, 2) == round( oc.available, 2), "Occupancy percentages are incosistent"
print('Occupancy calculated with available days is consistent with the one in table')

Occupancy calculated with available days is consistent with the one in table


## All Variables of full-sized JSON

|Column|Type|Description|
|------|----|-----------|
|url|str|
|name|str|
|stars|float|
|numberOfGuests|int|
|address|str|e.g., 'El Poblado, Medellín, Antioquia, Colombia'|
|roomType|str|
|location|dict|e.g., {'lat': 6.20859, 'lng': -75.56774} |
|reviews|list|metadata with reviews information|
|pricing|dict|pricing information, metadata|
|photos|list| URL of photos|
|primaryHost|dict| info of Host, metadata|
|additionalHosts|list|additional info of Host, metadata|
|isHostedBySuperhost|bool||
|isAvailable|float|1. or nan|
|calendar|list|Every day availability of listing|
|occupancyPercentage|float||
|listingAmenities|list| amenities in the premise|
|rootAmenitySections|list| XXX I do not know|
|seeAllAmenitySections|list||
|bathroomLabel|str|Numero de baños, `'2,5 baños'`|
|bedLabel|str|e.g., `'7 camas'`|
|bedroomLabel|str|e.g., `'3 dormitorios'`|
|hasHouseRules|bool|`True` or `False`|
|id|int|7-8 digit integer|
|isNewListing|bool|
|license|float|i do not know|
|listingRooms|list| type of bed in each room, metadata|
|p3SummaryAddress|str|Seems the same as `address`|
|roomTypeCategory|str| 'entire_home', 'private_room', 'hotel_room', 'shared_room'|
|sectionedDescription|dict|Text, `'authorType', 'description', 'houseRules'...`|
|tierId|int|All Zero, 0|
|guestControls|dict|`'allowsChildren', 'allowsEvents', 'allowsPets', ...`|
|hasSpecialOffer|bool|Mostly `False`|
|maxNights|int||
|minNights|int|
|priceDetails|list|Other expenses `'Gastos de limpieza'` etc.|
|hasWeWorkLocation|bool|All `False`| 
|isBusinessTravelReady|bool|All 'False'|
|hometourRooms|list|???|
|hometourSections|list|???|
|descriptionLocale|str|`'en', 'es', 'it', 'de', ...`|
|initialDescriptionAuthorType|str|All `'human'`| 
|localizedCheckInTimeWindow|str||
|localizedCheckOutTime|str|
|city|str|`'Medellín', 'Amagá', 'El Retiro', 'Envigado', ...`|
|country|str|`'Coloimbia'`|
|countryCode|str|`'Co'`|
|hasHostGuidebook|bool|All `False`|
|hasLocalAttractions|bool|All `False`|
|neighborhoodCommunityTags|list| ????|
|state|str|`'Antioquia', 'Antioquia, Ciudad de Bolivar ', ...`|
|paidGrowthRemarketingListingIds|list| ???|
|hasCommercialHostInfo|bool||
|reviewDetailsInterface|dict|`'reviewCount'` and `'reviewSummary'`|
|reviewsOrder|str|All `'language_country'`|
|nearbyAirportDistanceDescriptions|list|???|
|renderTierId|int|All 0|
|isHotel|bool|
|showReviewTag|bool|All `False`|
|isRepresentativeInventory|bool|???|
|localizedCity|str|`'Medellín', 'Amagá', 'El Retiro', ...`|
|highlights|list|Messages from the visitors|
|highlightsImpressionId|str|something like `'ebcd798b-56b9-245b-9663-269436fe77ef'`|
|categorizedPreviewAmenities|list|
|pointOfInterests|list|Empty ???|
|chinaPointsOfInterestMatcha|dict|Empty >???|
|pageViewType|str|All `'DEFAULT'`|
|previewTags|list|Empty ???|
|seeAllHometourSections|list|Empty ???|
|enableHighlightsVoting|bool|All `False`|
|listingExpectations|list|Some details to take note, e.g., 'noise', 'escaleras'|
|localizedListingExpectations|list|Similar to previous one|
|heroModule|dict|Empty ???|
|reviewsModule|dict|Inside `'localizedOverallRating'`|
|availabilityModule|dict|Empty ???|
|additionalHouseRules|str|Long text, rules|
|sortedReviews|list| Empty ???|
|documentDisplayPictures|list| Empty ???|
|sections|list| Empty ???|
|metadata|dict|Empty ???|
|p3ImpressionId|str| like `ebcd798b-56b9-245b-9663-269436fe77ef`
|idStr|int|Like `40751388`|
|paidGrowthRemarketingListingIdsStr|list|Empty ???|

In [165]:
data.paidGrowthRemarketingListingIdsStr

0       []
1       []
2       []
3       []
4       []
        ..
1993    []
1994    []
1995    []
1996    []
1997    []
Name: paidGrowthRemarketingListingIdsStr, Length: 1998, dtype: object